File: increasingring/src/main/java/com/baybaka/increasingring/config/RingerConfig.java
package com.baybaka.increasingring.config;

public class RingerConfig {
    private int startSoundLevel;
    private int interval;
    private int allowedMaxVolume;

    private boolean vibrateFirst;
    private int vibrateTimes;

    private boolean muteFirst;
    private int muteTimes;

    private boolean useMusicStream;

    public boolean isUseMusicStream() {
        return useMusicStream;
    }

    public void setUseMusicStream(boolean useMusicStream) {
        this.useMusicStream = useMusicStream;
    }


    public RingerConfig() {
        interval = 5;
    }

    public int getStartSoundLevel() {
        return startSoundLevel;
    }

    public void setInterval(int interval) {
        this.interval = interval;
    }

    public int getInterval() {
        return interval;
    }

    public int getAllowedMaxVolume() {
        return allowedMaxVolume;
    }

    public boolean isVibrateFirst() {
        return vibrateFirst;
    }

    public boolean isMuteFirst() {
        return muteFirst;
    }

    public void setStartSoundLevel(int startSoundLevel) {
        this.startSoundLevel = startSoundLevel;
    }

    public void setAllowedMaxVolume(int allowedMaxVolume) {
        this.allowedMaxVolume = allowedMaxVolume;
    }

    public void setVibrateFirst(boolean vibrateFirst) {
        this.vibrateFirst = vibrateFirst;
    }

    public void setMuteFirst(boolean muteFirst) {
        this.muteFirst = muteFirst;
    }

    public int getVibrateTimes() {
        return vibrateTimes;
    }

    public void setVibrateTimes(int vibrateTimes) {
        this.vibrateTimes = vibrateTimes;
    }

    public int getMuteTimes() {
        return muteTimes;
    }

    public void setMuteTimes(int muteTimes) {
        this.muteTimes = muteTimes;
    }
}

File: increasingring/src/main/java/com/baybaka/increasingring/config/RingerConfig.kt
package com.baybaka.increasingring.config

import android.media.AudioManager
import com.baybaka.increasingring.settings.SettingsService

class RingerConfig {
    var startSoundLevel: Int = 0
    var interval: Int = 5
    var allowedMaxVolume: Int = 0

    var isVibrateFirst: Boolean = false
    var vibrateTimes: Int = 0

    var isMuteFirst: Boolean = false
    var muteTimes: Int = 0

    var doNotRing = false
    var useMusicStream: Boolean = false

    override fun toString(): String {
        return "RingerConfig(startSoundLevel=$startSoundLevel, interval=$interval, allowedMaxVolume=$allowedMaxVolume, isVibrateFirst=$isVibrateFirst, vibrateTimes=$vibrateTimes, isMuteFirst=$isMuteFirst, muteTimes=$muteTimes, useMusicStream=$useMusicStream)"
    }

    fun update(settingsService: SettingsService) {
        updateStream(settingsService.soundStream)

        interval = settingsService.interval

        isMuteFirst = settingsService.isMuteFirst
        muteTimes = settingsService.muteTimesCount

        isVibrateFirst = settingsService.isVibrateFirst
        vibrateTimes = settingsService.vibrateTimesCount

        doNotRing = settingsService.isSkipRing
    }

    private fun updateStream(soundStream: Int) {
        useMusicStream = soundStream == AudioManager.STREAM_MUSIC
    }
}


